src/ucs/datastruct/lru.c

```diff
@@ -0,0 +1,145 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2001-2020. ALL RIGHTS RESERVED.
+* Copyright (C) Los Alamos National Security, LLC. 2019 ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "khash.h"
+#include "list.h"
+#include "lru.h"
+#include "ucs/debug/memtrack_int.h"
+
+#include <stdint.h>
+
+
+typedef struct {
+    void           *key;
+    ucs_list_link_t list;
+} ucs_lru_element_t;
+
+
+__KHASH_TYPE(lru_hash, uint64_t, ucs_lru_element_t);

```

Comment:
There are internal `khash` macros. Why do you use them?
If there is absolute justification - need to reconsider internal/exposed of `khash.h`.